src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -0,0 +1,816 @@
+        mock_app_mngr_,
+        request_controller_,
+        &mock_protocol_handler_,
+        &mock_hmi_handler_,
+        mock_command_holder_,
+        mock_rpc_protection_manager_,
+        hmi_so_factory_,
+        mobile_so_factory_);
+  }
+
+  MessageSharedPtr CreateMessage(
+      const smart_objects::SmartType type = smart_objects::SmartType_Null) {
+    return std::make_shared<smart_objects::SmartObject>(type);
+  }
+  void PrepareBasicMessage(MessageSharedPtr& message) {
+    (*message)[am::strings::params][am::strings::function_id] =
+        static_cast<mobile_apis::FunctionID::eType>(
+            mobile_apis::FunctionID::RESERVED);
+    (*message)[am::strings::params][am::strings::correlation_id] =
+        kCorrelationId;
+    (*message)[am::strings::params][am::strings::protocol_type] =
+        static_cast<int>(am::commands::CommandImpl::mobile_protocol_type_);
+    (*message)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+  }
+
+  // This class is created to allow access to protected method
+  // InitFunctionSchema from MOBILE_API
+  class MobileAPIWithPublicSchemaSetting : public mobile_apis::MOBILE_API {
+   public:
+    void SetSchema(mobile_apis::FunctionID::eType function_id,
+                   mobile_apis::messageType::eType message_type) {
+      InitFunctionSchema(function_id, message_type);
+    }
+  };
+
+  // This class is created to allow access to protected method
+  // InitFunctionSchema from HMI_API
+  class HMIAPIWithPublicSchemaSetting : public hmi_apis::HMI_API {
+   public:
+    void SetSchema(hmi_apis::FunctionID::eType function_id,
+                   hmi_apis::messageType::eType message_type) {
+      InitFunctionSchema(function_id, message_type);
+    }
+  };
+
+ protected:
+  hmi_apis::HMI_API hmi_so_factory_;
+  mobile_apis::MOBILE_API mobile_so_factory_;
+  testing::NiceMock<MockApplicationManager> mock_app_mngr_;
+  testing::NiceMock<MockRequestControlerSettings> mock_request_controler_;
+  testing::NiceMock<MockProtocolHandler> mock_protocol_handler_;
+  am::request_controller::RequestController request_controller_;
+  testing::NiceMock<MockHMIMessageHandler> mock_hmi_handler_;
+  testing::NiceMock<MockCommandHolder> mock_command_holder_;
+  std::shared_ptr<am::MockRPCProtectionManager> mock_rpc_protection_manager_;
+  std::shared_ptr<rpc_service::RPCService> rpc_service_;
+  std::shared_ptr<MockApplication> mock_app_ptr_;
+  am::MockMessageHelper& mock_message_helper_;
+  MockAppServiceManager mock_app_service_nmgr_;
+  testing::NiceMock<am::plugin_manager::MockRPCPluginManager>
+      mock_rpc_plugin_manager_;
+  testing::NiceMock<am::plugin_manager::MockRPCPlugin> mock_rpc_plugin_;
+  testing::NiceMock<MockCommandFactory> mock_command_factory_;
+};
+
+TEST_F(RPCServiceImplTest, ManageMobileCommand_MessageIsNullPtr_False) {
+  MessageSharedPtr message;
+  ASSERT_FALSE(rpc_service_->ManageMobileCommand(
```

Comment:
ASSERT usually used for preparation, not a check. I would advise using EXPECT_FALSE. 
Please also check for similar cases in this file.